src/modules/CRUD/components/Details/Read/AttachmentView/CategorySelector/CategorySelector.js

```diff
@@ -1,84 +1,120 @@
-      nextValue = value === '' ? inputValue : value;
-      setSelectedCategory(getValueFromId(nextValue));
-      setSelectValue(value);
+      setCategoriesLoaded(true);
+      setOptions(prevOptions => [...prevOptions, ...categories]);
     }
+    fetchCategories();
+  }, [attachmentCategoriesEndpoint, getAttachmentCategories, isCategoriesLoaded, options, t]);
 
-    if (name === 'category-selector-other') {
-      setInputValue(value);
-      nextValue = value;
-      setSelectedCategory({ id: null, name: nextValue });
-    }
 
-    onChange(nextValue);
-  }, [getValueFromId, inputValue, onChange]);
+  const [selectedCategory, setSelectedCategory] = useState(null);
 
-  const handleCancel = useCallback(() => {
-    setIsSubmitted(false);
-    onSubmit({});
-  }, [onSubmit]);
+  const handleValueChange = useCallback(value => {
+    setSelectedCategory(value);
+  }, []);
 
   const handleSubmit =  useCallback(event => {
     event.preventDefault();
-    setIsSubmitted(true);
     onSubmit(selectedCategory);
   }, [onSubmit, selectedCategory]);
 
+  const createNewItemRenderer = (
+    query,
+    active,
+    handleClick,
+  ) => (
+    <MenuItem
+      icon="add"
+      text={t('CRUD.details.attachment.category.create', { name: query })}
+      active={active}
+      onClick={handleClick}
+      shouldDismissPopover={false}
+    />
+  );
 
-  const Select = ({ value }) => (
-    <HTMLSelect onChange={handleChange} name="category-selector" value={value}>
-      {options.map(({ id, name: categoryName }) => (
-        <option key={id} value={id}>{categoryName}</option>
-      ))}
-      <option value="">{t('CRUD.details.attachment.category.new')}</option>
-    </HTMLSelect>
+  const itemRenderer = (option, { handleClick }) => (
+    <MenuItem
+      key={option.id}
+      text={option.name}
+      onClick={handleClick}
+    />
   );
 
-  if (isSubmitted) {
-    return (
-      <div className="category-selector category-selector--submitted">
-        <FormGroup
-          inline
-          helperText={t('CRUD.details.attachment.category.helper')}
-        >
-          <Tag large onRemove={handleCancel}>{selectedCategory.name}</Tag>
-        </FormGroup>
-      </div>
-    );
-  }
+  const itemListRenderer = ({ filteredItems, renderItem, renderCreateItem }) => {
+    const filteredItemsByGroupTitle = filteredItems.reduce((acc, item) => {
+      if (acc[item.title]) {
+        acc[item.title].push(item);
+      } else {
+        acc[item.title] = [item];
+      }
+      return acc;
+    }, {});
+    if (Object.values(filteredItemsByGroupTitle).length) {
+      return Object.entries(filteredItemsByGroupTitle).map(([key, value]) => (
+        <Menu key={key}>
+          <MenuDivider title={key} />
+          {value.map(renderItem)}
+        </Menu>
+      ));
+    }
+    return <Menu>{renderCreateItem()}</Menu>;
+  };
+
+  const itemPredicate = (query, option, _index, exactMatch) => {
+    const normalizedName = option.name.toLowerCase();
```

Comment:
I recommand you use a diacritics remover like in other search (see  https://github.com/Terralego/terra-admin/blob/cbb446042adb44d67de75f19329b268ec3e5d754/src/modules/RA/Campaign/components/AddViewpoint.js for an example).

Reply:
Yes top, thx!